codec/amino.go

```diff
@@ -61,6 +61,16 @@ func MustMarshalJSONIndent(cdc *LegacyAmino, obj interface{}) []byte {
 	return bz
 }
 
+// RegisterAminoMsg first checks that the msgName is <40 chars
+// (else this would break ledger nano signing: https://github.com/cosmos/cosmos-sdk/issues/10870),
+// then registers the concrete msg type with amino.
+func RegisterAminoMsg(cdc *LegacyAmino, msg interface{}, msgName string) {
```

Comment:
```suggestion
func RegisterAminoMsg(cdc *LegacyAmino, msg sdk.Msg, msgName string) {
```

any reason you didn't use sdk.Msg?

Reply:
good catch! This is something we should use.

Reply:
this would cause an import cycle
instead we could put this function into some package inside codec instead? what do you think about `codec/msg`? any other suggestions?

Reply:
I would propose even `codec/legacy`